src/discussions-trigger.ts

```diff
@@ -0,0 +1,175 @@
+Please edit the title to include the name on npm inside square brackets. 
+
+E.g. 
+ - \`"[@typescript/vfs] Does not x, y"\`
+ - \`"Missing x inside [node]"\`
+ - \`"[express] Broken support for template types"\`
+`;
+
+const gotAReferenceMessage = (module: string, owners: string[]) => `Thanks for the discussion about "${module}", some useful links for everyone: [npm](https://www.npmjs.com/package/${module}) / etc
+
+Pinging the DT module owners: ${owners.join(", ")}.
+`;
+
+
+async function pingAuthorsAndSetUpDiscussion(discussion: Discussion) {
+    const aboutNPMRef = extractNPMReference(discussion);
+    if (!aboutNPMRef) {
+        // https://gist.github.com/smashwilson/311e1487ddb40a455fc54d294cc63ad4#adddiscussioncomment
+        await updateOrCreateMainComment(discussion, couldNotFindMessage);
+        return;
+    } else {
+        const message = gotAReferenceMessage(aboutNPMRef, []);
+        await updateOrCreateMainComment(discussion, message);
+        await addLabel(discussion, aboutNPMRef, `Discussions related to ${aboutNPMRef}`);
+        return;
+    }
+}
+
+async function updateDiscordWithRequest(discussion: Discussion) {
+    const discordWebhookAddress = process.env.DT_MODULE_REQ_DISCORD_WEBHOOK;
+    if (!discordWebhookAddress) throw new Error("DT_MODULE_REQ_DISCORD_WEBHOOK not set in ENV");
+
+    // https://birdie0.github.io/discord-webhooks-guide/discord_webhook.html
+    const webhook = {  content: `New DT Module requested:`, embeds: [ { title: discussion.title, url: discussion.html_url } ] };
+    await fetch(discordWebhookAddress, { method: "POST", body: JSON.stringify(webhook), headers: { "content-type": "application/json" } });
+}
+
+
+async function updateOrCreateMainComment(discussion: Discussion, message: string) {
+    const discussionComments = await getCommentsForDiscussionNumber(discussion.number);
+    const previousComment = discussionComments.find(c => c.author.login === "typescript-bot");
+    if (previousComment) {
+        await client.mutate(createMutation<any>("updateDiscussionComment" as any, { body: message, commentId: previousComment.id }));
+    } else {
+        await client.mutate(createMutation<any>("addDiscussionComment" as any, { body: message, discussionId: discussion.node_id }));
+    }
+}
+
+async function addLabel(discussion: Discussion, labelName: string, description?: string) {
+
+    const existingLabel = await getLabelByName(labelName);
+    if (existingLabel && existingLabel.name === labelName) {
+        await client.mutate(createMutation<any>("addLabelsToLabelable" as any, { labelableId: discussion.node_id, labelIds: [existingLabel.id] }));
+    } else {
+        const dtRepoID = "MDEwOlJlcG9zaXRvcnk2MDkzMzE2";
+        // https://docs.github.com/en/graphql/reference/input-objects#createlabelinput
+        const color = crypto.randomBytes(Math.ceil(6/2)).toString("hex").slice(0,6);
```

Comment:
Won't it be easier to: `sha256(labelName).substring(0,6)`?

Reply:
It would! Doing so